cmake/list_insert_options.cmake

```diff
@@ -0,0 +1,55 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
```

Comment:
> What do you expect to happen if they export different values for the same key and why?

As we discussed it should print a warning (in any case).
It might even better to issue a  fatal error, as the situation is unresolvable (?).
This way it doesn't matter which comes first or last ;)